crates/ruma-common/src/api/metadata.rs

```diff
@@ -598,27 +613,46 @@ impl FromStr for MatrixVersion {
 impl MatrixVersion {
     /// Checks whether a version is compatible with another.
     ///
-    /// A is compatible with B as long as B is equal or less, so long as A and B have the same
-    /// major versions.
-    ///
-    /// For example, v1.2 is compatible with v1.1, as it is likely only some additions of
-    /// endpoints on top of v1.1, but v1.1 would not be compatible with v1.2, as v1.1
-    /// cannot represent all of v1.2, in a manner similar to set theory.
-    ///
-    /// Warning: Matrix has a deprecation policy, and Matrix versioning is not as
-    /// straight-forward as this function makes it out to be. This function only exists
-    /// to prune major version differences, and versions too new for `self`.
+    /// Currently, all versions of Matrix are considered backwards compatible with all the previous
+    /// versions, so this is equivalent to `self >= other`. This behaviour may change in the future,
+    /// if a new release is considered to be breaking compatibility with the previous ones.
     ///
-    /// This (considering if major versions are the same) is equivalent to a `self >= other`
-    /// check.
+    /// > ⚠ Matrix has a deprecation policy, and Matrix versioning is not as straightforward as this
+    /// > function makes it out to be. This function only exists to prune breaking changes between
+    /// > versions, and versions too new for `self`.
     pub fn is_superset_of(self, other: Self) -> bool {
-        let (major_l, minor_l) = self.into_parts();
-        let (major_r, minor_r) = other.into_parts();
-        major_l == major_r && minor_l >= minor_r
+        self >= other
```

Comment:
Why did you change the implementation? Do you think we'll actually remember to change it back if and when Matrix 2.0 happens? I doubt it.

Reply:
Matrix 2.0 will be a superset of whatever the last 1.XX will be. If this returns `false`, with the current `Metadata` functions it means that Matrix 2.0 won't use the current endpoints.

Reply:
Oh, alright. I wonder what we even need `into_parts`, `from_parts` for still. But I guess they're not flagged as dead code.

Reply:
We need them for the `metadata!` macro to do comparisons in const contexts.